peachjam/timeline_email_service.py

```diff
@@ -12,10 +16,34 @@
     send_saved_search_email_alert,
 )
 
+log = logging.getLogger(__name__)
+
 
 class TimelineEmailService:
+    @staticmethod
+    def already_alerted_today(user, event_type):
+        last_24_hrs = timezone.now() - timedelta(hours=24)
+        email_sent = TimelineEvent.objects.filter(
+            email_alert_sent_at__gte=last_24_hrs,
+            event_type=event_type,
+            user_following__user=user,
+        ).exists()
+        if email_sent:
+            log.info(
+                "%s email for %s has been sent within the last 24hrs: %s",
+                event_type,
+                user,
+                last_24_hrs,
+            )
+            return True
+        return False
+
     @staticmethod
     def send_email_alerts():
+        if not settings.PEACHJAM["EMAIL_ALERTS_ENABLED"]:
+            log.info("email alerts are disabled")
+            return
+
```

Comment:
@longhotsummer i've added the setting for disabling email alerts. I considered adding a setting in the Settings model, to have more control but we can add that if or when needed